state/indexer/block/kv/util.go

```diff
@@ -51,6 +51,51 @@ func heightKey(height int64) ([]byte, error) {
 	)
 }
 
+func keyBelongsToHeightRange(key []byte, left, right int64) bool {
+	// left included, right excluded
+	eventHeight, err := parseHeightFromEventKey(key)
+	if err == nil {
+		return left <= eventHeight && eventHeight < right
+	}
+
+	var (
+		blockHeightKeyPrefix string
+		possibleHeight       int64
+	)
+	remaining, err := orderedcode.Parse(string(key), &blockHeightKeyPrefix, &possibleHeight)
+	if err != nil {
+		return false
+	}
+	if len(remaining) == 0 &&
+		blockHeightKeyPrefix == types.BlockHeightKey &&
+		left <= possibleHeight &&
+		possibleHeight < right {
+		return true
+	}
+	return false
```

Comment:
```suggestion
	return len(remaining) == 0 &&
		blockHeightKeyPrefix == types.BlockHeightKey &&
		left <= possibleHeight &&
		possibleHeight < right
```
Just a suggestion, up to you

Reply:
Done